src/transport/PeerConnections.h

```diff
@@ -59,7 +59,7 @@ class PeerConnections
 
         for (size_t i = 0; i < kMaxConnectionCount; i++)
         {
-            if (!mStates[i].GetPeerAddress().IsInitialized())
+            if (!mStates[i].IsInitialized())
```

Comment:
In previous code, scalling this would make the connection 'initialized'.

With changes in IsInitialixed, this only seems to set mPeerAddress however does not change mPeerNodeId/keyId/localKeyId.

Could we pass in more data to the create function, to make sure that creation creates and marks as 'in use'?

I see the new CreateNewPeerConnectionState with extra args does just that. In this case, why did we keep this old version?

Alternatively maybe we should create separate initialization levels to query: if we need to create a connection without it being fully set (no keys set), then this method should take that into account and not allow overwriting of connections.

Reply:
@andy31415 with secure pairing PR, this flavor of function (initializing using just the IP address) will be obsoleted. I didn't remove it here, since the current code base is still using it.

Reply:
Could we remove it part of this PR? Or make existing code use the new function with dummy settings that mark the connection initialized?

Last resort, we could at least mark as [deperecated] with TODO to remove at least?

Reply:
@andy31415 I sent you some message in IM chat. Let me know if that helps answer this comment.

Reply:
So yes, I misread the IsInitialized code. Added approval.